edx_sga/static/js/src/edx_sga.js

```diff
@@ -1,22 +1,240 @@
 /* Javascript for StaffGradedAssignmentXBlock. */
 function StaffGradedAssignmentXBlock(runtime, element) {
+    function xblock($, _) {
+        var uploadUrl = runtime.handlerUrl(element, 'upload_assignment');
+        var downloadUrl = runtime.handlerUrl(element, 'download_assignment');
+        var annotatedUrl = runtime.handlerUrl(element, 'download_annotated');
+        var getStaffGradingUrl = runtime.handlerUrl(element, 'get_staff_grading_data');
+        var staffDownloadUrl = runtime.handlerUrl(element, 'staff_download');
+        var staffAnnotatedUrl = runtime.handlerUrl(element, 'staff_download_annotated');
+        var staffUploadUrl = runtime.handlerUrl(element, 'staff_upload_annotated');
+        var enterGradeUrl = runtime.handlerUrl(element, 'enter_grade');
+        var removeGradeUrl = runtime.handlerUrl(element, 'remove_grade');
+        var template = _.template($(element).find("#sga-tmpl").text());
+        var gradingTemplate;
 
-    function updateCount(result) {
-        $('.count', element).text(result.count);
-    }
+        function render(state) {
+            // Add download urls to template context
+            state.downloadUrl = downloadUrl;
+            state.annotatedUrl = annotatedUrl;
+            state.error = state.error ? state.error : false;
+
+            // Render template
+            var content = $(element).find("#sga-content").html(template(state));
+
+            // Set up file upload
+            $(content).find(".fileupload").fileupload({
+                url: uploadUrl,
+                add: function(e, data) {
+                    var do_upload = $(content).find(".upload").html('');
+                    $('<button/>')
+                        .text('Upload ' + data.files[0].name)
+                        .appendTo(do_upload)
+                        .click(function() {
+                            do_upload.text("Uploading...");
+                            data.submit();
+                        });
+                },
+                progressall: function(e, data) {
+                    var percent = parseInt(data.loaded / data.total * 100, 10);
+                    $(content).find(".upload").text(
+                        "Uploading... " + percent + "%");
```

Comment:
Hard coded English.  Take a look at ORA2 for a reference on how to get all strings in templates / python modules / javascript i18n'd.